cmd/azure-config-credentials-injector/main.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	clientIDEnvKey     = "AZURE_CLIENT_ID"
+	clientSecretEnvKey = "AZURE_CLIENT_SECRET"
+
+	clientIDCloudConfigKey     = "aadClientId"
+	clientSecretCloudConfigKey = "aadClientSecret"
+)
+
+var (
+	injectorCmd = &cobra.Command{
+		Use:   "azure-config-credentials-injector [OPTIONS]",
+		Short: "Cloud config credentials injection tool for azure cloud platform",
+		RunE:  mergeCloudConfig,
+	}
+
+	injectorOpts struct {
+		cloudConfigFilePath string
+		outputFilePath      string
+	}
+)
+
+func init() {
+	injectorCmd.PersistentFlags().AddGoFlagSet(flag.CommandLine)
+	injectorCmd.PersistentFlags().StringVar(&injectorOpts.cloudConfigFilePath, "cloud-config-file-path", "/tmp/cloud-config/cloud.conf", "Location of the original cloud config file.")
+	injectorCmd.PersistentFlags().StringVar(&injectorOpts.outputFilePath, "output-file-path", "/tmp/merged-cloud-config/cloud.conf", "Location of the generated cloud config file with injected credentials.")
+	err := injectorCmd.MarkFlagRequired("cloud-config-file-path")
+	if err != nil {
+		log.Fatal(err)
+	}
+	err = injectorCmd.MarkFlagRequired("output-file-path")
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Some spacing here might make this a bit easier to reason with
```suggestion
	
	if err := injectorCmd.MarkFlagRequired("cloud-config-file-path"); err != nil {
		log.Fatal(err)
	}
	
	if err := injectorCmd.MarkFlagRequired("output-file-path"); err != nil {
		log.Fatal(err)
	}
```

Reply:
I just understood that these flags are not required, and we can delete these checks.